ironfist/src/cpp/adventure/adv.cpp

```diff
@@ -2,6 +2,57 @@
 #include "adventure/map.h"
 #include "game/game.h"
 #include "scripting/hook.h"
+#include "prefs.h"
+
+extern void __fastcall NormalDialog(char *, int, int, int, int, int, int, int, int, int);
+
+int advManager::ProcessDeSelect(struct tag_message *GUIMessage_evt, int *a3, class mapCell * *a4) {
+	if(GUIMessage_evt->yCoordOrFieldID == 4)
+		{
+		DWORD hero_reminder_reg_dword = read_pref<DWORD>("Show Hero Movement Reminder");
+		//default is true, but read_pref() returns -1 if the value is not set
+		bool show_hero_movement_reminder = !(hero_reminder_reg_dword == 0);
+		
+		if(!show_hero_movement_reminder || !gpCurPlayer->HasMobileHero() ||
+				(
+				NormalDialog("One or more heroes may still move, are you sure you want to end your turn?",
+							2, -1, -1, -1, 0, -1, 0, -1, 0),
+				gpWindowManager->buttonPressedCode != 30726)

```

Comment:
I'm not sure what this is really telling me:
```
//
//Detected button
//fieldID : 30726
//	field_12 : 8
//	field_14 : 2
//	flags : 11468806
//	offsetY : 106
//	width : 98
//	height : 25
//	normalImageIdx : 7
//	armedImageIdx : 8
//	field_28 : 0
//	field_2A : 1
//	icnFileID : -478270621
```

We have no way to tell what the original name was from the binary, so we just have to make up something.

```
const int BUTTON_CODE_OKAY = 30726;
```